src/NCronJob/Configuration/Builder/DependentJobRegistryEntry.cs

```diff
@@ -2,7 +2,6 @@ namespace NCronJob;
 
 internal sealed class DependentJobRegistryEntry
 {
-    public required Type PrincipalType { get; init; }
     public List<JobDefinition> RunWhenSuccess { get; init; } = [];
```

Comment:
Could this life directly on the parent JobDefinition?

I don't see much of a need for `DepdentJobRegistryEntry` (and friends) anymore.

The question is how we can model something like:
```csharp
AddNcronJob(s => {
  s.AddJob<PrincipalJob>(...).ExecuteWhen(r => r.RunJob<DependentJob>());
  s.AddJob<DependentJob>(r => r.WithCronExpression(...)).ExecutenWhen(r => r.RunJob<DepDepJob>());
```

So if `PrincipalJob` runs, `DependentJob` should run afterward and `DepDepJob` after that.

Reply:
> Could this life directly on the parent JobDefinition?
> 
> I don't see much of a need for `DepdentJobRegistryEntry` (and friends) anymore.

If that's ok with you, I'd rather work on this in a follow up PR as an internal refactoring.

This one is already a bit "stuffy".

Reply:
Agreed.
> Does the test do what you'd expect?
Yes